### PR TITLE
[Modules] Storage Accounts : Add SFTP feature to storage account module # 2378

### PR DESCRIPTION
**Pull request by karthikvenkat17**
# Description

Updated the storage account resource module to enable SFTP and NFSv3 for blob storage
Added localUsers sub module to deploy SFTP users

## Pipeline references

[Storage: StorageAccounts](https://github.com/karthikvenkat17/CARML/actions/workflows/ms.storage.storageaccounts.yml)

# Type of Change

- [x] New feature (non-breaking change which adds functionality)

# Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: modules/Microsoft.Storage/storageAccounts/localUsers/deploy.bicep [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+@description('Required. Indicates whether ssh password exists. Set it to false to remove existing SSH password.')
+param hasSshPassword bool
+
+@description('Optional. The local user home directory.')
+param homeDirectory string = ''
+
+@description('Required. The permission scopes of the local user.')
+param permissionScopes array
+
+@description('Optional. The local user ssh authorized keys for SFTP.')
```
Comment (ahmadabdalla) [inline modules/Microsoft.Storage/storageAccounts/localUsers/deploy.bicep:23]:
```suggestion
@description('Optional. The local user SSH authorized keys for SFTP.')
```

File: modules/Microsoft.Storage/storageAccounts/localUsers/deploy.bicep [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+@description('Conditional. The name of the parent Storage Account. Required if the template is used in a standalone deployment.')
+param storageAccountName string
+
+@description('Required. The local username to be used for SFTP Authentication.')
+param name string
+
+@description('Optional. Indicates whether shared key exists. Set it to false to remove existing shared key.')
+param hasSharedKey bool = false
+
+@description('Required. Indicates whether ssh key exists. Set it to false to remove existing SSH key.')
```
Comment (ahmadabdalla) [inline modules/Microsoft.Storage/storageAccounts/localUsers/deploy.bicep:11]: In the description field only, please change `ssh` to `SSH` . Applicable to other sections this document

**Review by ahmadabdalla — CHANGES_REQUESTED** [review]
Thank you for your contribution @karthikvenkat17 . Please see comments.

File: modules/Microsoft.Storage/storageAccounts/localUsers/deploy.bicep [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    homeDirectory: homeDirectory
+    permissionScopes: permissionScopes
+    sshAuthorizedKeys: empty(sshAuthorizedKeys) ? null : sshAuthorizedKeys
+  }
+}
+
+@description('The name of the local user created for SFTP Authentication.')
+output name string = localUsers.name
+
+@description('The resource group of the deployed management policy.')
```
[the remark this answers is not in the document]
Comment (karthikvenkat17) [reply modules/Microsoft.Storage/storageAccounts/localUsers/deploy.bicep]: Updated as per comments

File: modules/Microsoft.Storage/storageAccounts/localUsers/deploy.bicep [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+@description('Optional. The local user home directory.')
+param homeDirectory string = ''
+
+@description('Required. The permission scopes of the local user.')
+param permissionScopes array
+
+@description('Optional. The local user ssh authorized keys for SFTP.')
+param sshAuthorizedKeys array = []
+
+
```
[the remark this answers is not in the document]
Comment (karthikvenkat17) [reply modules/Microsoft.Storage/storageAccounts/localUsers/deploy.bicep]: Resolved by formatting

File: modules/Microsoft.Storage/storageAccounts/deploy.bicep [hunk 5]
```diff
@@ -85,6 +85,15 @@ param minimumTlsVersion string = 'TLS1_2'
 @description('Optional. If true, enables Hierarchical Namespace for the storage account.')
 param enableHierarchicalNamespace bool = false
```
[the remark this answers is not in the document]
Comment (karthikvenkat17) [reply modules/Microsoft.Storage/storageAccounts/deploy.bicep:86]: Agreed, parameter changed to Conditional

File: modules/Microsoft.Storage/storageAccounts/localUsers/deploy.bicep (hunk 2 above)
Comment (karthikvenkat17) [reply modules/Microsoft.Storage/storageAccounts/localUsers/deploy.bicep]: Corrected in all sections

File: settings.yml [hunk 6]
```diff
@@ -123,4 +123,4 @@ variables:
 #     value: 'OtherVersion'
 #   - name: preferredAzurePowerShellVersion
 #     value: '4.4.0'
-#
+#
```
Comment (ahmadabdalla) [inline settings.yml:126]: @karthikvenkat17 .. Everything is OK. Just one final comment. Please remove unnecessary commit to the `settings.yml`. Just a matter of restoring the file to the content from main.

Comment (karthikvenkat17) [reply settings.yml]: @ahmadabdalla Sorry that was an oversight on my part. Its restored now.

**Review by ahmadabdalla — APPROVED** [review]
👍🏼

**Pull request merged**